instrumentation/aws-sdk/aws-sdk-2.2/testing/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/AbstractAws2ClientTest.java

```diff
@@ -0,0 +1,708 @@
+            "GetObject",
+            "GET",
+            (Function<S3Client, Object>)
+                c ->
+                    c.getObject(
+                        GetObjectRequest.builder().bucket("somebucket").key("somekey").build()),
+            (Function<S3AsyncClient, Future<?>>)
+                c ->
+                    c.getObject(
+                        GetObjectRequest.builder().bucket("somebucket").key("somekey").build(),
+                        AsyncResponseTransformer.toBytes()),
+            "1234567890"));
+  }
+
+  @ParameterizedTest
+  @MethodSource("provideS3Arguments")
+  void testS3SendOperationRequestWithBuilder(
+      String operation, String method, Function<S3Client, Object> call) throws Exception {
+    S3ClientBuilder builder = S3Client.builder();
+    if (Boolean.getBoolean("testLatestDeps")) {
+      Method forcePathStyleMethod =
+          S3ClientBuilder.class.getMethod("forcePathStyle", Boolean.class);
+      forcePathStyleMethod.invoke(builder, true);
+    }
+    configureSdkClient(builder);
+    S3Client client =
+        builder
+            .endpointOverride(clientUri)
+            .region(Region.AP_NORTHEAST_1)
+            .credentialsProvider(CREDENTIALS_PROVIDER)
+            .build();
+
+    server.enqueue(HttpResponse.of(HttpStatus.OK, MediaType.PLAIN_TEXT_UTF_8, ""));
+
+    Object response = call.apply(client);
+
+    assertThat(response.getClass().getSimpleName())
+        .satisfiesAnyOf(
+            v -> assertThat(v).startsWith(operation),
+            v -> assertThat(response).isInstanceOf(ResponseInputStream.class));
+    clientAssertions("S3", operation, method, response, "UNKNOWN");
+  }
+
+  @ParameterizedTest
+  @MethodSource("provideS3Arguments")
+  void testS3AsyncSendOperationRequestWithBuilder(
+      String operation,
+      String method,
+      Function<S3Client, Object> call,
+      Function<S3AsyncClient, Future<?>> asyncCall,
+      String body)
+      throws NoSuchMethodException,
+          InvocationTargetException,
+          IllegalAccessException,
+          ExecutionException,
+          InterruptedException {
+    S3AsyncClientBuilder builder = S3AsyncClient.builder();
+    if (Boolean.getBoolean("testLatestDeps")) {
+      Method forcePathStyleMethod =
+          S3AsyncClientBuilder.class.getMethod("forcePathStyle", Boolean.class);
+      forcePathStyleMethod.invoke(builder, true);
+    }
+    configureSdkClient(builder);
+    S3AsyncClient client =
+        builder
+            .endpointOverride(clientUri)
+            .region(Region.AP_NORTHEAST_1)
+            .credentialsProvider(CREDENTIALS_PROVIDER)
+            .build();
+
+    server.enqueue(HttpResponse.of(HttpStatus.OK, MediaType.PLAIN_TEXT_UTF_8, body));
+
+    Future<?> response = asyncCall.apply(client);
+    response.get();
+
+    clientAssertions("S3", operation, method, response, "UNKNOWN");
+  }
+
+  @Test
+  void testKinesisSendOperationRequestWithBuilder() throws Exception {
```

Comment:
doesn't really throw Exception, `throws Exception` could be removed from some other methods too